### DISCUSSION TITLE
Feature Request: Allow parameters as a viable target file path when exporting results with COPY ... TO

### DISCUSSION THREAD

**Discussion by Eulium** [duckdb/duckdb#14830] [Ideas]
### Request: 
**Allow parameters as a viable target file path when exporting results with COPY ... TO**
Currently COPY ... TO requires a  constant as the export file target as defined in 
```
| COPY '(' SelectStmt ')' TO opt_program copy_file_name opt_with copy_options
```
```
copy_file_name:
            Sconst                                    { $$ = $1; }
            | STDIN                                    { $$ = NULL; }
            | STDOUT                                { $$ = NULL; }
            | IDENT '.' ColId                        { $$ = psprintf("%s.%s", $1, $3); }
            | IDENT                                    { $$ = $1; }

        ;
```
this prevents to use of COPY TO  in prepared statement in which the target file name is given as a parameter or as the result of an expression. 

Changing this behavior to support expressions or parameters would enable the inclusion of COPY TO inside of prepared statements and further improve readability of SQL scripts.  
**Example:** 
```
PREPARE query_person AS
COPY (
    SELECT *
    FROM person
    WHERE starts_with(name, $2)
      AND age >= $1;
) 
TO $3 ; 

EXECUTE query_person(35, 'A', 'A_35.csv');
EXECUTE query_person(40, 'B', 'B_40.csv');
```

**Comment by tgs** [comment]
I would also really like this!

I'm using DuckDB to create Parquet files from some calculations. The filename I'm supposed to write to comes from outside my code. I'm not really worried about the filename being intentional SQL injection, it's more like a temp file name, but I still get twitchy when I have to use string interpolation like `f"COPY (...) TO '{out}'"`. I would prefer to use a DuckDB named parameter, but it doesn't work:

```
>>> import duckdb
>>> conn = duckdb.connect()
>>> conn.execute("COPY (select 1, 2, 3) TO $out (FORMAT parquet)", {'out': 'out.pq'})
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    conn.execute("COPY (select 1, 2, 3) TO $out (FORMAT parquet)", {'out': 'out.pq'})
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
duckdb.duckdb.ParserException: Parser Error: syntax error at or near "$"
```

Could we have $variable support there? Or is there a reliable way to render a python string to a quoted DuckDB string?

Thank you for DuckDB by the way, it's running my `lag(blah) OVER (PARTITION BY thing)` query almost 1000x faster than SQL Server, and it's far more convenient when getting data in and out!

**Comment by Mytherin** [comment]
This has been implemented in https://github.com/duckdb/duckdb/pull/18795

**Reply by tgs** [reply]
Wonderful, thank you so much!
